src/Interpolators.f90

```diff
@@ -0,0 +1,146 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief A Fortran 2003 module implementing interpolation routines for input
+!>        arrays returning either interpolated values or a flat "extrapolation"
+!>        when interpolation is out of range.
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
```

Comment:
maybe add some detail here clarifying that it's limited to 3D or smaller

Reply:
These details have been added.